pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -211,34 +164,31 @@ func getClusterConfigFromSecret(k8s kubernetes.Interface,
 // validateCluster validate a cluster
 func validateCluster(cluster *k8s_cr.Cluster) (err error) {
 	if cluster.TypeMeta.Kind != "Cluster" {
-		err = multierr.Append(err, fmt.Errorf("bad kind in configuration: `%s` != 'Cluster'", cluster.TypeMeta.Kind))
-	}
-
-	if cluster.ObjectMeta.Annotations[ClusterPilotEndpoint] == "" {
-		err = multierror.Append(err, fmt.Errorf("cluster %s doesn't have a valid pilot endpoint", cluster.ObjectMeta.Name))
-	}
-
-	switch serviceregistry.ServiceRegistry(cluster.ObjectMeta.Annotations[ClusterPlatform]) {
-	case serviceregistry.KubernetesRegistry:
-	case serviceregistry.ConsulRegistry:
-	case serviceregistry.EurekaRegistry:
-	case serviceregistry.CloudFoundryRegistry:
-	default:
-		err = multierror.Append(err, fmt.Errorf("cluster %s has unsupported platform %s",
-			cluster.ObjectMeta.Name, cluster.ObjectMeta.Annotations[ClusterPlatform]))
-	}
-
-	if cluster.ObjectMeta.Annotations[ClusterPilotCfgStore] != "" {
-		if _, err1 := strconv.ParseBool(cluster.ObjectMeta.Annotations[ClusterPilotCfgStore]); err1 != nil {
-			err = multierror.Append(err, err1)
+		err = multierror.Append(err, fmt.Errorf("bad kind in configuration: `%s` != 'Cluster'", cluster.TypeMeta.Kind))
+	}
+	// Default is k8s.
+	if len(cluster.ObjectMeta.Annotations[ClusterPlatform]) > 0 {
+		switch serviceregistry.ServiceRegistry(cluster.ObjectMeta.Annotations[ClusterPlatform]) {
+		// Currently only supporting kubernetes registry,
+		case serviceregistry.KubernetesRegistry:
+		case serviceregistry.ConsulRegistry:
+			fallthrough
+		case serviceregistry.EurekaRegistry:
+			fallthrough
+		case serviceregistry.CloudFoundryRegistry:
+			fallthrough
+		default:
+			err = multierror.Append(err, fmt.Errorf("cluster %s has unsupported platform %s",
+				cluster.ObjectMeta.Name, cluster.ObjectMeta.Annotations[ClusterPlatform]))
 		}
 	}
+
 	if cluster.ObjectMeta.Annotations[ClusterAccessConfigSecret] == "" {
-		err = multierror.Append(err, fmt.Errorf("cluster %s doesn't have a valid config secret", cluster.ObjectMeta.Name))
-	} else {
-		if cluster.ObjectMeta.Annotations[ClusterAccessConfigSecretNamespace] == "" {
-			cluster.ObjectMeta.Annotations[ClusterAccessConfigSecretNamespace] = "istio-system"
-		}
+		// by default, expect a secret with the same name as the cluster
+		cluster.ObjectMeta.Annotations[ClusterAccessConfigSecretNamespace] = cluster.Name
```

Comment:
I think using just secret name is dangerous as it could cause unpleasant collisions. I would still suggest to use labels as an idnicator that secret carries cluster config. It does not have to be done in this PR, once it is merged I will push   a change to address it.

Reply:
That's going away with the config changes.